### PR TITLE
Turn high availability support into a feature gate

### PR DESCRIPTION
**Pull request by dtantsur**
Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

**Comment by iurygregory** [comment]
lgtm

File: main.go [hunk 1]
```diff
@@ -128,6 +139,13 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	if err := metal3iov1alpha1.CurrentFeatureGate.SetFromMap(featureGates); err != nil {
```
Comment (MahnoorAsghar) [inline main.go:142]: Wouldnt the featureGates variable be empty here?

Comment (dtantsur) [reply main.go:142]: It's loaded from the environment and the CLI flags on lines 124-132 (line 124 binds it to the flag).

Comment (MahnoorAsghar) [reply main.go:142]: So when you call `featureGatesFlag.Set(defaultFeatureGates)` on line 126, it makes changes to `featureGates` as well?

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**